Upgrade dependencies to work with Node v22+

anoadragon453 opened the pull request:
Fixes a bunch of flagged packages.

Codex CLI was used to do most of this. The only commit I don't fully understand is:

- [Update webpack dev server](https://github.com/matrix-org/fosdem-schedule-element-widget/commit/24c7fc25546eee82ff24ecc9043b3bede79d57a9)
    - `config/webpack.config.js` appears to be mainly updating to new config option names
    - `config/webpackDevServer.config.js` appears to be mostly boilerplate

Recommended to review commit-by-commit.

anoadragon453 committed:
Run 'yarn upgrade'

anoadragon453 committed:
Fix eslint warnings

Required to fix the build.

anoadragon453 committed:
Bump http-proxy-middleware from ^1.0.6 to ^2.0.7

Clears a high vuln in the older version.

anoadragon453 committed:
Pass `NODE_OPTIONS=--openssl-legacy-provider`

to fix `yarn build` on Node 18+

anoadragon453 committed:
Update webpack dev server

Removes a bunch of transitive vulns

anoadragon453 committed:
Update react and related utilities

Use `resolutions` to pin transitive dependencies:

- The `@types/react*` packages needed to be pinned to v17, else we'd get
  v19, which caused typing errors.
- The rest needed to be pinned to avoid vulns.

anoadragon453 committed:
Remove unused `react-scripts` `portable-fetch` deps

anoadragon453 committed:
Bump `@babel/core` from `7.12.3` to `^7.24.7` to fix tests

Was previously failing with:

```
 FAIL  src/App.test.tsx
  ● Test suite failed to run

    [BABEL] /home/work/code/fosdem-schedule-element-widget/src/setupTests.ts: Requires Babel "^7.22.0 || >8.0.0-alpha <8.0.0-beta", but was loaded with "7.12.3". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel. (While processing: "/home/work/code/fosdem-schedule-element-widget/node_modules/@babel/plugin-syntax-import-attributes/lib/index.js")

      at Object.assertVersion (node_modules/@babel/core/lib/config/helpers/config-api.js:74:15)
      at builder (node_modules/@babel/plugin-syntax-import-attributes/src/index.ts:8:7)
      at node_modules/@babel/helper-plugin-utils/src/index.ts:64:12
      at node_modules/@babel/core/lib/config/full.js:211:14
      at Function.<anonymous> (node_modules/@babel/core/lib/gensync-utils/async.js:26:3)
      at evaluateSync (node_modules/gensync/index.js:251:28)
      at cachedFunction (node_modules/@babel/core/lib/config/caching.js:68:46)
          at cachedFunction.next (<anonymous>)
      at loadPluginDescriptor (node_modules/@babel/core/lib/config/full.js:247:42)
          at loadPluginDescriptor.next (<anonymous>)
      at loadPluginDescriptors (node_modules/@babel/core/lib/config/full.js:158:30)
          at loadPluginDescriptors.next (<anonymous>)
      at node_modules/@babel/core/lib/config/full.js:186:21
      at loadFullConfig (node_modules/@babel/core/lib/config/full.js:169:5)
          at loadFullConfig.next (<anonymous>)
      at transform (node_modules/@babel/core/lib/transform.js:25:45)
          at transform.next (<anonymous>)
```

reivilibre reviewed (approved):
If it's tested to work in both dev & docker build, then I accept this, but the webpack config is virtually unreviewable to me. Mercifully it's just a build tool.

I find myself in awe at the amount of stuff needed to build a schedule widget :sweat_smile:. If I was dealing with this more often I'd probably suggest switching to Vite, but it's not worth the hassle atm

anoadragon453 committed:
Bump node to 24 LTS in Dockerfile

anoadragon453 committed:
fix linting error

You need to store the result in a variable before exporting it, apparently.

anoadragon453 committed:
Fix lack of `application/json` Content-Type header

Prevent login from working during testing. It's possible this isn't
added automatically anymore, or matrix.org just requires it now.

Either way, needed to happen!

anoadragon453 commented:
Tested after a node bump in the Dockerfile, and it looks good!

<img width="2256" height="1468" alt="image" src="https://github.com/user-attachments/assets/2ecba83e-f5fe-4955-ad99-15d304d004ea" />

anoadragon453 committed:
Bump NodeJS version in CI to 24

anoadragon453 committed:
Bump actions/checkout to v6.0.1 and pin

while we're here

anoadragon453 committed:
Bump docker/login-action to v3.6.0 and pin
